### PR TITLE
Feature/l5 c search in modal screen

### PR DESCRIPTION
**Pull request by Khikar-1997**

### CODE REVIEW

File: src/Components/ContactList/FavoriteContactInfo.js [hunk 1]
```diff
@@ -3,23 +3,21 @@ import { Text, View, StyleSheet } from 'react-native';
...
     width: 79,
     height: 24,
     marginLeft: 10,
-    borderBottomColor: colors.grey,
-    marginBottom: 25,
+    borderBottomColor: colors.quartz,
+    marginBottom: 50,
     alignItems: 'center',
   },
   contactNameAndSurnameText: {
```
Comment (AnushavanGhulyan) [inline src/Components/ContactList/FavoriteContactInfo.js:20]: add "textAlign", and remove parent view

File: src/Screens/Contacts.screen.js [hunk 2]
```diff
@@ -26,7 +41,23 @@ const styles = StyleSheet.create({
...
+  headerLeft: () => (
+    <Avatar
+      name="Khikar"
+      surname="Harutyunyan"
+      image={imagePaths.homePageAvatarImage}
+      size="small"
+    />
+  ),
+};
+
```
Comment (AnushavanGhulyan) [inline src/Screens/Contacts.screen.js]:
```suggestion
Contacts.options = ({ navigation }) => ({
headerRight: () => (
        <Button
          onPress={() => navigation.navigate(Routes.ADD_CONTACTS)}
          icon="add"
          size="small"
        />
      ),
  headerLeft: () => (
    <Avatar
      name="Khikar"
      surname="Harutyunyan"
      image={imagePaths.homePageAvatarImage}
      size="small"
    />
  ),
});

```

File: src/Screens/Contacts.screen.js [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
+    navigation.setOptions({
+      headerRight: () => (
+        <Button
+          onPress={() => navigation.navigate(Routes.ADD_CONTACTS)}
+          icon="add"
+          size="small"
+        />
+      ),
+    });
+  });
```
Comment (AnushavanGhulyan) [inline src/Screens/Contacts.screen.js]: see below

File: src/Screens/SearchContact.screen.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  const navigation = useNavigation();
+
+  const [searchedText, setSearchedText] = useState('');
+  const MOCK_DATA = route.params.contacts;
+
+  const handleChange = e => {
+    e.persist();
+    clearTimeout(timeoutRef.current);
+    timeoutRef.current = setTimeout(() => {
+      console.log('1');
```
Comment (AnushavanGhulyan) [inline src/Screens/SearchContact.screen.js]: remove console. log, and try without e.persist

File: src/Screens/SearchContact.screen.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      setSearchedText(e.nativeEvent.text);
+    }, 2000);
```
Comment (AnushavanGhulyan) [inline src/Screens/SearchContact.screen.js]: decrease delay to 250ms

File: src/Screens/SearchContact.screen.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        textColor: colors.logan,
+        hideWhenScrolling: false,
+      },
+      headerLeft: () => (
+        <Button
+          onPress={() => navigation.navigate(Routes.CONTACTS)}
+          icon="arrow"
+          size="small"
+        />
+      ),
```
Comment (AnushavanGhulyan) [inline src/Screens/SearchContact.screen.js]: same here

File: src/Screens/SearchContact.screen.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    });
+  }, [navigation]);
+
+  return (
+    <View style={styles.searchedContactsContainer}>
+      <SearchedContacts value={searchedText} contacts={MOCK_DATA} />
+    </View>
```
Comment (AnushavanGhulyan) [inline src/Screens/SearchContact.screen.js]: remove view

File: src/UIKit/Button/index.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Text, TouchableOpacity } from 'react-native';
+import Icons from '../Icons';
+import buttonSize from './size';
+import buttonColor from './variant';
+
+const Button = ({ onPress, icon, title = '', size, variant = '' }) => (
```
Comment (AnushavanGhulyan) [inline src/UIKit/Button/index.js]: Add "default" variant,

File: src/UIKit/Button/index.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Text, TouchableOpacity } from 'react-native';
+import Icons from '../Icons';
+import buttonSize from './size';
+import buttonColor from './variant';
```
Comment (AnushavanGhulyan) [inline src/UIKit/Button/index.js]: Move these styles into this file, and merge it in one stylesheet object

File: src/Screens/SearchContact.screen.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+  const [searchedText, setSearchedText] = useState('');
+  const MOCK_DATA = route.params.contacts;
+
+  const handleChange = e => {
+    e.persist();
+    clearTimeout(timeoutRef.current);
+    timeoutRef.current = setTimeout(() => {
+      console.log('1');
+      setSearchedText(e.nativeEvent.text);
```
Comment (AnushavanGhulyan) [inline src/Screens/SearchContact.screen.js]: Move all logic here , filter data here, and passed filtered data to list

**Review by AnushavanGhulyan — CHANGES_REQUESTED** [review]

**Review by griarman — APPROVED** [review]

**Pull request merged**